### PR TITLE
4 server initialization

### PR DESCRIPTION
**Pull request by Hekard**
- Added gin project with zap-logger and golang-migrate
- Added jwt token middleware

### CODE REVIEW

File: app-server/internal/repository/db.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func PerformMigration(db *sql.DB) {
+	driver, err := postgres.WithInstance(db, &postgres.Config{})
+	if err != nil {
+		log.Fatal(err)
+	}
+	migrationsDir := "/app/app-server/internal/repository/migrations"
+	m, err := migrate.NewWithDatabaseInstance(
+		"file://"+migrationsDir,
```
Comment (Justyanna) [inline app-server/internal/repository/db.go:38]: Formatowanko

File: app-server/internal/mock/mockTokenHandler.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	"go.uber.org/zap"
+)
+
+func (h *BaseMockHandler) MockTokenHandler(c *gin.Context, secret_byte []byte) {
+	logger := c.MustGet("logger").(*zap.SugaredLogger)
+
+	expirationTime := time.Now().Add(12 * time.Hour)
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"userID": "bar",
+		"exp":    expirationTime.Unix(),
```
Comment (Justyanna) [inline app-server/internal/mock/mockTokenHandler.go:18]: Formatowanko

File: app-server/internal/auth/authMiddleware.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+	"go.uber.org/zap"
+)
+
+func AuthMiddleware(secret_byte []byte) gin.HandlerFunc {
```
Comment (Justyanna) [inline app-server/internal/auth/authMiddleware.go:14]: To bym podzieliła na mniejsze funkcje

**Review by abraham-kaczmarski — CHANGES_REQUESTED** [review]
Good start, we need to agree on some conventions.

File: app-server/internal/utilis/environment.go
Comment (abraham-kaczmarski) [inline app-server/internal/utilis/environment.go:1]: Fix package name.

File: app-server/internal/utilis/environment.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package utilis
+
+import "os"
+
+func getSecretKey() []byte {
```
Comment (abraham-kaczmarski) [inline app-server/internal/utilis/environment.go:5]: What about other getters?

**Review by abraham-kaczmarski — CHANGES_REQUESTED** [review]
Just some more cleanup.

**Review by abraham-kaczmarski — APPROVED** [review]
LGTM

**Pull request merged**